src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -403,22 +403,88 @@ private static Version GetMSBuildVersionInPath()
         /// <returns>The msbuild directory.</returns>
         public static string GetMsbuildDirectory(string userVersion, IConsole console)
         {
-            List<Toolset> installedToolsets;
-            using (var projectCollection = new ProjectCollection())
+            // Try to find msbuild for mono from hard code path.
+            if (RuntimeEnvironmentHelper.IsMono)
```

Comment:
We should add a test that exercises this when environment variable `OSTYPE` starts/contains with `darwin`. Maybe try out: http://stackoverflow.com/questions/394230/detect-the-os-from-a-bash-script